SteamKit2/SteamKit2/Steam/SteamClient/SteamClient.cs

```diff
@@ -376,24 +373,21 @@ protected override bool OnClientMsgReceived( IPacketMsg? packetMsg )
             }
 
             // pass along the clientmsg to all registered handlers
-            foreach ( DictionaryEntry kvp in handlers )
+            foreach ( var value in handlers )
```

Comment:
This loop is the important code which is called a lot, and looking at OrderedDictionary implementation of enumerators, I don't think it's very efficient.